---
bool has_litter
bool success